delphi-frontend/src/test/java/au/com/integradev/delphi/cfg/ControlFlowGraphTest.java

```diff
@@ -0,0 +1,1324 @@
+            block(element(NameReferenceNode.class, "Foo")).succeedsTo(1),
+            block(element(NameReferenceNode.class, "I"))
+                .branchesTo(2, 0)
+                .withTerminator(ForInStatementNode.class)));
+  }
+
+  @Test
+  void testForInBreak() {
+    test(
+        "for I in List do Break;",
+        checker(
+            block(element(NameReferenceNode.class, "List")).succeedsTo(1),
+            terminator(StatementTerminator.BREAK).jumpsTo(0, 1),
+            block(element(NameReferenceNode.class, "I"))
+                .branchesTo(2, 0)
+                .withTerminator(ForInStatementNode.class)));
+  }
+
+  @Test
+  void testForInConditionalBreak() {
+    test(
+        "for I in List do if I = 1 then Break;",
+        checker(
+            block(element(NameReferenceNode.class, "List")).succeedsTo(1),
+            block(
+                    element(NameReferenceNode.class, "I"),
+                    element(IntegerLiteralNode.class, "1"),
+                    element(BinaryExpressionNode.class)
+                        .withCheck(binaryOpTest(BinaryOperator.EQUAL)))
+                .branchesTo(2, 1)
+                .withTerminator(IfStatementNode.class),
+            terminator(StatementTerminator.BREAK).jumpsTo(0, 1),
+            block(element(NameReferenceNode.class, "I"))
+                .branchesTo(3, 0)
+                .withTerminator(ForInStatementNode.class)));
+  }
+
+  @Test
+  void testForInContinue() {
+    test(
+        "for I in List do Continue;",
+        checker(
+            block(element(NameReferenceNode.class, "List")).succeedsTo(1),
+            terminator(StatementTerminator.CONTINUE).jumpsTo(1, 1),
+            block(element(NameReferenceNode.class, "I"))
+                .branchesTo(2, 0)
+                .withTerminator(ForInStatementNode.class)));
+  }
+
+  @Test
+  void testForInConditionalContinue() {
+    test(
+        "for I in List do if I = 1 then Continue;",
+        checker(
+            block(element(NameReferenceNode.class, "List")).succeedsTo(1),
+            block(
+                    element(NameReferenceNode.class, "I"),
+                    element(IntegerLiteralNode.class, "1"),
+                    element(BinaryExpressionNode.class)
+                        .withCheck(binaryOpTest(BinaryOperator.EQUAL)))
+                .branchesTo(2, 1)
+                .withTerminator(IfStatementNode.class),
+            terminator(StatementTerminator.CONTINUE).jumpsTo(1, 1),
+            block(element(NameReferenceNode.class, "I"))
+                .branchesTo(3, 0)
+                .withTerminator(ForInStatementNode.class)));
+  }
+
+  @Test
+  void testBreakOutsideOfLoop() {
+    assertThatThrownBy(() -> test("Break;", checker()))
+        .withFailMessage("'Break' statement not in loop statement.")
+        .isInstanceOf(IllegalStateException.class);
+  }
+
+  @Test
+  void testContinueOutsideOfLoop() {
+    assertThatThrownBy(() -> test("Continue;", checker()))
```

Comment:
SonarLint:

> Refactor the code of the lambda to have only one invocation possibly throwing a runtime exception.

This `checker()` invocation should be moved out of the lambda into a separate variable that is then captured.